api/v1/model/src/commonMain/kotlin/Repository.kt

```diff
@@ -33,7 +39,16 @@ data class Repository(
 
     /** The url to the repository. */
     val url: String
-)
+) {
+    companion object {
+        const val INVALID_URL_MESSAGE = "The entity url is malformed."
+        const val USER_INFO_MESSAGE = "The entity url must not contain userinfo."
+
+        fun isValidUrl(url: String): Boolean = runCatching { URL(url) }.isSuccess
```

Comment:
Note that constructing a `URL` only accepts a limited set of protocols and might actually make network requests. If this is just about a syntax check, use `URI.create()` instead.

Reply:
Thanks for the hint, I didn't know the limitations of `URL`. I replaced it with `URI.create()`.